server/src/main/java/org/apache/uniffle/server/netty/ShuffleServerNettyHandler.java

```diff
@@ -101,6 +102,12 @@ public void exceptionCaught(Throwable cause, TransportClient client) {
   public void handleSendShuffleDataRequest(TransportClient client, SendShuffleDataRequest req) {
     RpcResponse rpcResponse;
     String appId = req.getAppId();
+    StatusCode status = verifyRequest(appId);
+    if (status != StatusCode.SUCCESS) {
+      rpcResponse = new RpcResponse(req.getRequestId(), status, Strings.EMPTY);
```

Comment:
You need to call `releaseNettyBufferAndMetrics` which was introduced in https://github.com/apache/incubator-uniffle/pull/1918 to avoid memory leaks. 

cc @maobaolong

Reply:
Done

Reply:
You just revert the code. You mean it can fast fail in `cacheShuffleData`, so we don't need this anymore, right?

Reply:
> You just revert the code. You mean it can fast fail in `cacheShuffleData`, so we don't need this anymore, right?

Right.